### PR TITLE
fix(appsync-modelgen-plugin): minor fix for null safety

### PR DESCRIPTION
**Pull request by AaronZyLee**
#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-codegen/blob/master/CONTRIBUTING.md#tests)
- [x] Relevant documentation is changed or added (and PR referenced)
- [x] Breaking changes to existing customers are released behind a feature flag or major version update
- [x] Changes are tested using sample applications for all relevant platforms (iOS/android/flutter/Javascript) that use the feature added/modified

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by haverchuck** [comment]
@AaronZyLee @renebrandel @Ashish-Nanda 

There should be two small changes for `amplify-codegen/packages/amplify-codegen/src/commands/models.js`:

line 82: 
```diff
      context.print.warning(
-        'Generating Dart Models with null safety. To opt out of null safe models, turn off the “enabledartnullsafety” feature flag.',
+.       'Generating Dart Models with null safety. To opt out of null safe models, turn off the “enableDartNullSafety” feature flag. Learn more: https://docs.amplify.aws/lib/project-setup/null-safety/q/platform/flutter'.
      );
```

line 87:
```diff
      context.print.warning(
-        'Generating Dart Models without null safety. To generate null safe data models, turn on the “enabledartnullsafety” feature flag and set your Dart SDK version to “>= 2.12.0”.',
+.       'Generating Dart Models without null safety. To generate null safe data models, turn on the “enabledartnullsafety” feature flag and set your Dart SDK version to “>= 2.12.0”. Learn more: https://docs.amplify.aws/lib/project-setup/null-safety/q/platform/flutter',
      );
```

@Ashish-Nanda @renebrandel Please confirm.

File: packages/amplify-codegen/src/commands/models.js [hunk 1]
```diff
@@ -80,12 +80,12 @@ async function generateModels(context) {
     context.print.warning(`Detected feature flag: “enabledartnullsafety : ${enableDartNullSafety}”`);
     if (isMinimumDartVersionSatisfied && enableDartNullSafety) {
       context.print.warning(
-        'Generating Dart Models with null safety. To opt out of null safe models, turn off the “enabledartnullsafety” feature flag.',
+        'Generating Dart Models with null safety. To opt out of null safe models, turn off the “enabledartnullsafety” feature flag. Learn more: https://docs.amplify.aws/lib/project-setup/null-safety/q/platform/flutter',
```
Comment (haverchuck) [inline packages/amplify-codegen/src/commands/models.js:83]:
```suggestion
        'Generating Dart Models with null safety. To opt out of null safe models, turn off the “enableDartNullSafety” feature flag. Learn more: https://docs.amplify.aws/lib/project-setup/null-safety/q/platform/flutter',
```

File: packages/amplify-codegen/src/commands/models.js [hunk 2]
```diff
@@ -80,12 +80,12 @@ async function generateModels(context) {
...
       );
     } else {
       enableDartNullSafety = false;
       context.print.warning(
-        'Generating Dart Models without null safety. To generate null safe data models, turn on the “enabledartnullsafety” feature flag and set your Dart SDK version to “>= 2.12.0”.',
+        'Generating Dart Models without null safety. To generate null safe data models, turn on the “enabledartnullsafety” feature flag and set your Dart SDK version to “>= 2.12.0”. Learn more: https://docs.amplify.aws/lib/project-setup/null-safety/q/platform/flutter',
```
Comment (haverchuck) [inline packages/amplify-codegen/src/commands/models.js:88]:
```suggestion
        'Generating Dart Models without null safety. To generate null safe data models, turn on the “enableDartNullSafety” feature flag and set your Dart SDK version to “>= 2.12.0”. Learn more: https://docs.amplify.aws/lib/project-setup/null-safety/q/platform/flutter',
```

**Comment by haverchuck** [comment]
Also need change in `validateDartSDK.js`:

line 9:
```diff
  try {
    const config = yaml.load(fs.readFileSync(path.join(projectRoot, PUBSPEC_FILE_NAME), 'utf8'));
    const version = semver.minVersion(config.environment.sdk);
    if (semver.satisfies(version, '>= 2.12.0')) {
-      context.print.warning('\nDetected Dart SDK version: “>= 2.12.0” is “true”');
+      context.print.warning('\nDetected Dart SDK version 2.12.0 or above.');
      return true;
    }
-    context.print.warning('\nDetected Dart SDK version: “>= 2.12.0” is “false”.');
+    context.print.warning('\nDetected Dart SDK version below 2.12.0.');
    return false;
  } catch (e) {
-    context.print.warning('\nCould not detect Dart SDK version, defaulting to “>= 2.12.0”.');
+    context.print.warning('\nCould not detect Dart SDK version, defaulting to 2.12.0.');
    return true;
  }
```

File: packages/amplify-codegen/src/utils/validateDartSDK.js [hunk 3]
```diff
@@ -10,13 +10,13 @@ function validateDartSDK(context, projectRoot) {
...
-      context.print.warning('\nDetected Dart SDK version: “>= 2.12.0” is “true”');
+      context.print.warning('\nDetected Dart SDK version 2.12.0 or above.');
       return true;
     }
-    context.print.warning('\nDetected Dart SDK version: “>= 2.12.0” is “false”.');
+    context.print.warning('\nDetected Dart SDK version below 2.12.0.');
     return false;
   } catch (e) {
-    context.print.warning('\nCould not detect Dart SDK version, defaulting to “>= 2.12.0”.');
+    context.print.warning('\nCould not detect Dart SDK version, defaulting to 2.12.0.');
```
Comment (Ashish-Nanda) [inline packages/amplify-codegen/src/utils/validateDartSDK.js:19]: Should we remove the full stop at the end of these statements? It looks like another decimal point.
Also we don't seem to have them in the other warnings.

Comment (haverchuck) [reply packages/amplify-codegen/src/utils/validateDartSDK.js:19]: Makes sense to me - I updated the suggestion.

**Review by fjnoyp — APPROVED** [review]

**Pull request merged**